### PR TITLE
Create azure_ad_azurehound_discovery.yml

### PR DESCRIPTION
**Pull request by blueteam0ps**
Adding AzureHound detection via default UA string.

### CODE REVIEW

File: rules/cloud/azure/azure_ad_azurehound_discovery.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+tags:
+    - attack.t1087.004
+    - attack.t1526
+    - attack.discovery
+logsource:
+    product: azure
+    service: signinlogs
+detection:
+    selection_ua:
+        userAgent|starswith: 'azurehound'
```
Comment (nasbench) [inline rules/cloud/azure/azure_ad_azurehound_discovery.yml:18]:
```suggestion
        userAgent|startswith: 'azurehound'
```

File: rules/cloud/azure/azure_ad_azurehound_discovery.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        ResultType: '0'
```
Comment (nasbench) [inline rules/cloud/azure/azure_ad_azurehound_discovery.yml:19]:
```suggestion
        ResultType: 0
```

**Review by nasbench — CHANGES_REQUESTED** [review]

File: rules/cloud/azure/azure_ad_azurehound_discovery.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+date: 2022/11/27
+tags:
+    - attack.t1087.004
+    - attack.t1526
+    - attack.discovery
+logsource:
+    product: azure
+    service: signinlogs
+detection:
+    selection_ua:
```
Comment (frack113) [inline rules/cloud/azure/azure_ad_azurehound_discovery.yml:17]:
```suggestion
    selection:
```
as condition use selection

**Review by frack113 — CHANGES_REQUESTED** [review]

File: rules/cloud/azure/azure_ad_azurehound_discovery.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+tags:
+    - attack.discovery
+    - attack.t1087.004
+    - attack.t1526
+logsource:
+    product: azure
+    service: signinlogs
+detection:
+    selection:
+        userAgent|startswith: 'azurehound'
```
[the remark this answers is not in the document]
Comment (blueteam0ps) [reply rules/cloud/azure/azure_ad_azurehound_discovery.yml:18]: Ye so the actual UA comes as azurehound followed by the version no . Hence starts with. But I get your point to have it match on any as it is a low FP UA string.

**Review by nasbench — APPROVED** [review]

**Review by frack113 — APPROVED** [review]

**Pull request merged**